website_mass_mailing_name/static/src/js/website_mass_mailing_name.js

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2016-2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define("website_mass_mailing_name.subscribe", function (require) {
+    "use strict";
+    require("mass_mailing.website_integration");
+    var animation = require("website.content.snippets.animation");
+
+    animation.registry.subscribe.include({
+        start: function(editable_mode) {
+            this.$email = this.$(".js_subscribe_email");
+            this.$name = this.$(".js_subscribe_name");
+            // Thanks upstream for your @$&#?!! inheritance-ready code.
+            // Injecting ajax events to modify behavior of snippet.
+            if (this.$name) {
+                $(document).ajaxSend($.proxy(this.on_ajax_send, this));
+            }
+            return this._super(editable_mode);
+        },
+
+        _onClick: function() {
+            function validateEmail(sEmail) {
+                var filter = /^([\w-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/;
```

Comment:
This regexp is wrong, sorry.

No matter what you could possibly imagine, you'll almost always fail email regexp validation. :laughing: Go http://emailregex.com and see for yourself.

However, the good news is that you won't need a regexp. See below. You'll be able to drop the `validateEmail()` function.

Reply:
My bad, ok I will change it asap